src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -451,45 +476,118 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
             }
         }
 
-        public static unsafe void ScaleU(float scale, Span<float> dst)
+        public static unsafe void Scale(float scale, Span<float> dst)
         {
-            fixed (float* pdst = dst)
+            fixed (uint* pLeadingAlignmentMask = &LeadingAlignmentMask[0])
+            fixed (uint* pTrailingAlignmentMask = &TrailingAlignmentMask[0])
+            fixed (float* pd = dst)
             {
-                float* pDstCurrent = pdst;
-                float* pEnd = pdst + dst.Length;
+                float* pdLim = pd + dst.Length;
 
+                int length = dst.Length;
                 Vector256<float> scaleVector256 = Avx.SetAllVector256(scale);
 
-                while (pDstCurrent + 8 <= pEnd)
+                if (length < 8)
                 {
-                    Vector256<float> dstVector = Avx.LoadVector256(pDstCurrent);
+                    for(int i = 0; i < length; i++)
```

Comment:
it would be nice to explode this out like I did in the Sum PR: <https://github.com/dotnet/machinelearning/pull/1112/files#diff-1b97f6bad8c8b4dba44e33671e40baf4R930>

It should provide slightly better perf for these really small inputs.